Automattic/wp-calypso
# Plans 2023: Update intro offers to be derived from either site-plans or plans endpoints

Related to https://github.com/Automattic/dotcom-forge/issues/4570
Branched from https://github.com/Automattic/wp-calypso/pull/84605

## Proposed Changes

- Updates `useIntroOffers` hook to extract intro offers from either `useSitePlans` or `usePlans` queries (corresponding to `/rest/v1.3/sites/[id]/plans` and `/rest/v1.5/plans` endpoints)
- Gives precedence to `useSitePlans` intro offers. So these will override anything coming from `usePlans`
- Returns a union, not an intersection (so plans returned from either endpoint will be included in result)
- The above functionality should be covered in the unit tests included

## Testing Instructions



* Go to `/plans` and `/start/plans` and ensure to regressions in pricing
* Go to `/plans/[new woo site]` and ensure the `$1` intro offers are shown for all the plans in the grid (ensure billing descriptions render correctly) - can get a new site from `/setup/wooexpress` 

## Pre-merge Checklist
